--- portmidi/CMakeLists.txt	2009-11-19 22:44:03.000000000 -0800
+++ portmidi_patched/CMakeLists.txt	2010-04-02 18:46:17.965463707 -0700
@@ -73,5 +73,4 @@
 add_subdirectory(pm_dylib)
 
 # Cannot figure out how to make an xcode Java application with CMake
-add_subdirectory(pm_java)
 
--- portmidi/pm_linux/Makefile	2009-01-19 20:01:54.000000000 -0800
+++ portmidi_patched/pm_linux/Makefile	2010-04-02 18:45:17.507663986 -0700
@@ -44,8 +44,7 @@
 current: all
 
 all: $(pmlib) $(ptlib) pm_test/test pm_test/sysex pm_test/midithread \
-	pm_test/latency pm_test/midithru pm_test/qtest pm_test/mm \
-	pm_java/pmjni/libpmjni.so pm_java/pmdefaults/pmdefaults.jar 
+	pm_test/latency pm_test/midithru pm_test/qtest pm_test/mm
 
 $(pmlib): pm_linux/Makefile $(pmobjects)
 	ar -cr $(pmlib) $(pmobjects)
@@ -88,67 +87,10 @@
 pm_test/qtest: pm_linux/Makefile pm_test/qtest.o $(pmlib) $(ptlib)
 	$(CC) pm_test/qtest.o -o pm_test/qtest $(pmlib) $(ptlib) $(ALSALIB)
 
-#------------ Java stuff here --------------
-
-pm_java/pmjni/libpmjni.so: pm_java/pmjni/pmjni.c \
-                       pm_java/pmjni/jportmidi_JportMidiApi.h
-	gcc -o $@ -shared -Wl,-soname,libpmjni.so \
-	    -lgcj -fPIC \
-            -Ipm_common -Iporttime -Ipm_java/pmjni \
-	    pm_java/pmjni/pmjni.c -static -lc $(pmlib) $(ptlib) $(ALSALIB)
-
-#pm_java/pmjni/pmjni.so: pm_java/pmjni/pmjni.o \
-#                        pm_java/pmjni/jportmidi_JportMidiApi.h
-#	cd pm_java/pmjni; gcc -fPIC -shared -o pmjni.so \
-#           -Wl,--whole-archive,-soname,pmjni.so \
-#           pmjni.o -lgcj -Wl,--no-whole-archive
-
-JAVASRC = pm_java/jportmidi/JPortMidi.java \
-          pm_java/jportmidi/JPortMidiException.java \
-          pm_java/jportmidi/JPortMidiApi.java
-
-pm_java/jportmidi/JPortMidi.class: $(JAVASRC)
-	javac pm_java/jportmidi/*.java
-
-pm_java/jportmidi/JPortMidiException.class: $(JAVASRC)
-	javac pm_java/jportmidi/*.java
-
-pm_java/jportmidi/JPortMidiApi.class: $(JAVASRC)
-	javac pm_java/jportmidi/*.java
-
-DEFAULTSSRC = pm_java/pmdefaults/PmDefaultsFrame.java \
-              pm_java/pmdefaults/PmDefaults.java
-
-pm_java/pmdefaults/PmDefaultsFrame.class: $(DEFAULTSRC)
-	cd pm_java; javac -classpath . pmdefaults/*.java
-
-pm_java/pmdefaults/PmDefaults.class: $(DEFAULTSRC)
-	cd pm_java; javac -classpath . pmdefaults/*.java
-
-
-DEFAULTCLASS = pm_java/jportmidi/JPortMidi.class \
-               pm_java/jportmidi/JPortMidiException.class \
-               pm_java/jportmidi/JPortMidiApi.class \
-               pm_java/pmdefaults/PmDefaultsFrame.class \
-               pm_java/pmdefaults/PmDefaults.class
-
-
-# make the jar file:
-#    work in pm_java directory
-#    temporarily copy logo image to current directory
-#    copy the interface library to the 
-pm_java/pmdefaults/pmdefaults.jar: $(DEFAULTCLASS)
-	cd pm_java; cp pmdefaults/portmusic_logo.png .
-	cd pm_java; jar cmf pmdefaults/manifest.txt pmdefaults.jar \
-                            pmdefaults/*.class portmusic_logo.png \
-                            jportmidi/*.class
-	cd pm_java; rm portmusic_logo.png
-	# just in case, make it executable
-	cd pm_java; chmod +x pmdefaults/pmdefaults
 
 install:
-	cp pm_java/pmjni/libpmjni.so /usr/local/lib
-	cp pm_java/pmdefaults.jar /usr/share/java
+	#cp pm_java/pmjni/libpmjni.so /usr/local/lib
+	#cp pm_java/pmdefaults.jar /usr/share/java
 	# do not chmod +x here since we are probably root
 	cp pm_java/pmdefaults/pmdefaults /usr/local/bin
 
--- portmidi/pm_common/CMakeLists.txt	2009-11-19 22:18:06.000000000 -0800
+++ portmidi_patched/pm_common/CMakeLists.txt	2010-04-02 18:48:47.046187730 -0700
@@ -72,11 +72,11 @@
     # JAVA_INCLUDE_PATH2; if no, then we need to make both JAVA_INCLUDE_PATH
     # and JAVA_INCLUDE_PATH2 set by user (will need clear documentation
     # because JAVA_INCLUDE_PATH2 is pretty obscure)
-    set(JAVA_INCLUDE_PATH  ${JAVA_INCLUDE_PATH-UNKNOWN}
-        CACHE STRING "where to find Java SDK include directory")
-    set(JAVA_INCLUDE_PATHS ${JAVA_INCLUDE_PATH} ${JAVA_INCLUDE_PATH}/linux)
+    #set(JAVA_INCLUDE_PATH  ${JAVA_INCLUDE_PATH-UNKNOWN}
+    #    CACHE STRING "where to find Java SDK include directory")
+    #set(JAVA_INCLUDE_PATHS ${JAVA_INCLUDE_PATH} ${JAVA_INCLUDE_PATH}/linux)
     # libjvm.so is found relative to JAVA_INCLUDE_PATH:
-    set(JAVAVM_LIB ${JAVA_INCLUDE_PATH}/../jre/lib/i386/client/libjvm.so)
+    #set(JAVAVM_LIB ${JAVA_INCLUDE_PATH}/../jre/lib/i386/client/libjvm.so)
 
     set(LINUXSRC pmlinuxalsa pmlinux finddefault)
     prepend_path(LIBSRC ../pm_linux/ ${LINUXSRC})
@@ -104,27 +104,18 @@
     # message(STATUS "JAVAVM_LIB: " ${JAVAVM_LIB})
   endif(WIN32)
 endif(UNIX)
-set(JNI_EXTRA_LIBS ${PM_NEEDED_LIBS} ${JAVAVM_LIB})
 
 # this completes the list of library sources by adding shared code
 list(APPEND LIBSRC pmutil portmidi)
 
 # now add the shared files to make the complete list of library sources
 add_library(portmidi-static ${LIBSRC})
-set_target_properties(portmidi-static PROPERTIES OUTPUT_NAME "portmidi_s")
+set_target_properties(portmidi-static PROPERTIES OUTPUT_NAME "portmidi")
 target_link_libraries(portmidi-static ${PM_NEEDED_LIBS})
 
-# define the jni library
-include_directories(${JAVA_INCLUDE_PATHS})
-
-set(JNISRC ${LIBSRC} ../pm_java/pmjni/pmjni.c)
-add_library(pmjni SHARED ${JNISRC})
-target_link_libraries(pmjni ${JNI_EXTRA_LIBS})
-set_target_properties(pmjni PROPERTIES EXECUTABLE_EXTENSION "jnilib")
-
 # install the libraries (Linux and Mac OS X command line)
 if(UNIX)
-  INSTALL(TARGETS portmidi-static pmjni
+  INSTALL(TARGETS portmidi-static
     LIBRARY DESTINATION /usr/local/lib
     ARCHIVE DESTINATION /usr/local/lib)
 # .h files installed by pm_dylib/CMakeLists.txt, so don't need them here
