--- a/mozpluggerrc	2011-03-03 22:40:50.000000000 +0100
+++ b/mozpluggerrc	2011-05-11 23:15:04.796501158 +0200
@@ -279,13 +279,13 @@
 application/x-pdf:pdf:PDF file
 text/pdf:pdf:PDF file
 text/x-pdf:pdf:PDF file
-	ACROREAD()
 	repeat noisy swallow(kpdf) fill: kpdf "$file"
-	repeat noisy swallow(Xpdf) fill: xpdf -g +9000+9000 "$file"
 	repeat noisy swallow(okular) fill: okular "$file"
         repeat noisy swallow(epdfview) fill: epdfview "$file"
+	repeat noisy swallow(evince) fill: evince "$file"
 	GV()
-	repeat noisy fill exits: evince "$file"
+	repeat noisy swallow(Xpdf) fill: xpdf -g +9000+9000 "$file"
+	ACROREAD()
 
 application/x-dvi:dvi:DVI file
         repeat swallow(kdvi) fill: kdvi "$file"
