--- lib/ffmpeg/avcodec.pas	2009-10-14 23:46:31.022129060 +0200
+++ lib/ffmpeg/avcodec.pas	2009-10-14 23:52:39.085038745 +0200
@@ -79,16 +79,6 @@
                             (LIBAVCODEC_MIN_VERSION_MINOR * VERSION_MINOR) +
                             (LIBAVCODEC_MIN_VERSION_RELEASE * VERSION_RELEASE);
 
-(* Check if linked versions are supported *)
-{$IF (LIBAVCODEC_VERSION < LIBAVCODEC_MIN_VERSION)}
-  {$MESSAGE Error 'Linked version of libavcodec is too old!'}
-{$IFEND}
-
-(* Check if linked version is supported *)
-{$IF (LIBAVCODEC_VERSION > LIBAVCODEC_MAX_VERSION)}
-  {$MESSAGE Error 'Linked version of libavcodec is not yet supported!'}
-{$IFEND}
-
 const
   AV_NOPTS_VALUE: cint64  = $8000000000000000;
   AV_TIME_BASE            = 1000000;
--- lib/ffmpeg/avformat.pas	2009-10-15 00:03:02.179003818 +0200
+++ lib/ffmpeg/avformat.pas	2009-10-15 00:04:55.021004476 +0200
@@ -79,16 +79,6 @@
                             (LIBAVFORMAT_MIN_VERSION_MINOR * VERSION_MINOR) +
                             (LIBAVFORMAT_MIN_VERSION_RELEASE * VERSION_RELEASE);
 
-(* Check if linked versions are supported *)
-{$IF (LIBAVFORMAT_VERSION < LIBAVFORMAT_MIN_VERSION)}
-  {$MESSAGE Error 'Linked version of libavformat is too old!'}
-{$IFEND}
-
-(* Check if linked versions are supported *)
-{$IF (LIBAVFORMAT_VERSION > LIBAVFORMAT_MAX_VERSION)}
-  {$MESSAGE Error 'Linked version of libavformat is not yet supported!'}
-{$IFEND}
-
 {$IF LIBAVFORMAT_VERSION >= 52020000} // 52.20.0
 (**
  * Returns the LIBAVFORMAT_VERSION_INT constant.
--- avutil.pas	2010-02-14 23:00:01.213732883 +0100
+++ avutil.pas	2010-02-14 22:59:18.844609123 +0100
@@ -85,6 +85,15 @@
                           (LIBAVUTIL_MIN_VERSION_MINOR * VERSION_MINOR) +
                           (LIBAVUTIL_MIN_VERSION_RELEASE * VERSION_RELEASE);
 
-(* Check if linked versions are supported *)
-{$IF (LIBAVUTIL_VERSION < LIBAVUTIL_MIN_VERSION)}
-  {$MESSAGE Error 'Linked version of libavutil is too old!'}
-{$IFEND}
-
-{$IF (LIBAVUTIL_VERSION > LIBAVUTIL_MAX_VERSION)}
-  {$MESSAGE Error 'Linked version of libavutil is not yet supported!'}
-{$IFEND}
-
 {$IF LIBAVUTIL_VERSION >= 49008000} // 49.8.0
 (**
  * Returns the LIBAVUTIL_VERSION_INT constant.
